src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -121,14 +120,17 @@ Task RaiseExceptionReceived(Exception e, string action)
                     concurrentSessionSemaphoreAquired = true;
 
                     await this.maxPendingAcceptSessionsSemaphoreSlim.WaitAsync(this.pumpCancellationToken).ConfigureAwait(false);
-                    session = await this.client.AcceptMessageSessionAsync().ConfigureAwait(false);
+                    IMessageSession session = await this.client.AcceptMessageSessionAsync().ConfigureAwait(false);
                     if (session == null)
                     {
                         await Task.Delay(Constants.NoMessageBackoffTimeSpan, this.pumpCancellationToken).ConfigureAwait(false);
                         continue;
                     }
 
-                    TaskExtensionHelper.Schedule(() => this.MessagePumpTaskAsync(session));
```

Comment:
In the current code, nothing would have failed since the pump would schedule it before the `session` would be overwritten in the next loop. 
But just changed it to keep it risk free.